src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -155,12 +155,9 @@ public boolean isSlice()
         }
 
         @Override
-        public boolean inRestrictionCanBeUsedWith(ColumnsExpression expression)
+        public boolean isSupportedByRestrictionsOn(ColumnsExpression expression)
         {
-            if (expression.kind() == ColumnsExpression.Kind.ELEMENT)
-                return expression.elementKind() != ElementExpression.Kind.COLLECTION_ELEMENT;
-
-            return true;
+            return expression.kind() != ColumnsExpression.Kind.ELEMENT;
```

Comment:
Good point, I think it will be great if we add here the same comment added in `Operator#validateFor`: 
`// this method is used only in restrictions, not in conditions where different rules apply for now`